src/ipc/server.rs

```diff
@@ -89,6 +89,8 @@ fn on_new_ipc_client(state: &mut State, stream: UnixStream) {
         }
     };
 
+    state.backend.refresh_ipc_outputs(&mut state.niri);
```

Comment:
Computing outputs on every request is something I wanted to avoid with this Arc-Mutex. If we compute outputs like this then we might as well drop the whole Arc-Mutex and do the same as `niri msg workspaces`.

That said, I'm not entirely sure yet how I want to do this. Currently `niri msg outputs` contains *mostly* purely backend-specific stuff (with the exception of logical output info), and this is adding a layout property to it. One way to solve this would be to keep the Arc-Mutex as is, then query only the layout properties at the time of handling the IPC request (i.e. with a channel like `niri msg workspaces`). Another way is to have a separate IPC call for the layout info. Haven't decided what would be the best yet.

Reply:
makes total sense, i can hop in there and try to reduce the frequency of refreshes.

Reply:
i was just thinking through these options, and sway includes a `focused` field in the `get_outputs` ipc response, which was my frame of reference. however, a `focused-output` call would semantically match the `focused-window` ipc message, so that might be the simplest solution at the moment.

Reply:
That would be the simplest I guess.